pkg/server/server.go

```diff
@@ -350,6 +434,10 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 
 // Close releases the server resources.
 func (s *Server) Close() {
+	if s.rateLimitedCheckResolverCloser != nil {
+		s.rateLimitedCheckResolverCloser()
+	}
```

Comment:
```suggestion
	if s.rateLimitedCheckResolver != nil {
		s.rateLimitedCheckResolver.Close()
	}
```

FWIW i also don't like that we have 

```go
if s.cachedCheckResolverCloser != nil {
	s.cachedCheckResolverCloser()
}
```

I know it's not your code to change, but i don't really like this new pattern that has started to creep where we save a pointer to a function unnecessarily.